errors/errors.go

```diff
@@ -47,19 +47,23 @@ func (p *Error) StackString() string {
 
 // VerboseString returns the error message, stack trace and contexts
 func (p *Error) VerboseString() string {
-	return fmt.Sprintf("%s\nPublic context: %+v\nPrivate context: %+v\n%s", p.Error(), p.PublicContext, p.PrivateContext, p.StackString())
+	return fmt.Sprintf("%s\nParams: %+v\n%s", p.Error(), p.Params, p.StackString())
 }
 
 // New creates a new error for you. Use this if you want to pass along a custom error code.
 // Otherwise use the handy shorthand factories below
-func New(code int, message string, context ...map[string]string) *Error {
-	return errorFactory(code, message, context...)
+func New(code string, message string, params map[string]string) *Error {
+	return errorFactory(code, message, params)
 }
 
 // Wrap takes any error interface and wraps it into an Error.
 // This is useful because an Error contains lots of useful goodies, like the stacktrace of the error.
 // NOTE: If `err` is already an `Error` the passed contexts will be ignored
-func Wrap(err error, context ...map[string]string) *Error {
+func Wrap(err error, params map[string]string) *Error {
```

Comment:
It would be better if this method returned `error` IMO, because otherwise you can't safely call `errors.Wrap(nil)` as illustrated here:

```
func blurb() error {
  return terrors.Wrap(nil)
}

if err := blurb(); err != nil {
  // this will now execute (I think)...
}
```

Addressing this would require rewriting a bit of code in Typhon (it used to be in `endpoint.go` right after executing the endpoint handler, but not sure where it is after the mercury/typhon refactor). Essentially, if you need a `*Error` after calling `errors.Wrap`, you'll have to cast it (which you can safely do afterwards). @obeattie what do you think?

Right now we have to be careful never to do this in service code:
```
result, err := someDatabaseOperation()
return result, errors.Wrap(err) // ALARM! can only do this if err != nil
```
Stackoverflow says this is bad design anyway, but it definitely caused bugs in the past